tests/accessor/local_accessor_api_common.h

```diff
@@ -0,0 +1,134 @@
+                                   "Check api for empty local_accessor")) {
+      queue
+          .submit([&](sycl::handler &cgh) {
+            AccT acc;
+            test_accessor_methods_common(acc, 0 /* expected_byte_size*/,
+                                         0 /*expected_size*/,
+                                         util::get_cts_object::range<dims>::get(
+                                             0, 0, 0) /*expected_range*/);
+          })
+          .wait_and_throw();
+    }
+
+    SECTION(get_section_name<dims>(type_name, access_mode_name, target_name,
+                                   "Check api for local_accessor")) {
+      T data(expected_val);
+      constexpr size_t global_range_size = 64;
+      constexpr size_t local_range_size = 2;
+      auto global_range = util::get_cts_object::range<dims>::get(
+          global_range_size, global_range_size, global_range_size);
+      auto local_range = util::get_cts_object::range<dims>::get(
+          local_range_size, local_range_size, local_range_size);
+      sycl::nd_range<dim> nd_range(global_range, local_range);
+      bool res = false;
+      {
+        sycl::buffer<T, dims> data_buf(&data, r);
+        sycl::buffer res_buf(&res, sycl::range(1));
+        queue
+            .submit([&](sycl::handler &cgh) {
+              AccT acc(local_range, cgh);
+              AccT acc_other(local_range, cgh);
+
+              test_accessor_methods_common(
+                  acc, sizeof(T) * local_range.size() /* expected_byte_size*/,
+                  local_range.size() /*expected_size*/,
+                  local_range /*expected_range*/);
+
+              sycl::accessor res_acc(res_buf, cgh);
+              cgh.parallel_for(nd_range, [=](sycl::nd_item<dim> item) {
+                auto ref_1 = acc[0];
+
+                auto id = util::get_cts_object::id<dims>::get(1, 1, 1);
+                auto ref_2 = get_subscript_overload<T, AccT, dims>(acc, id);
+
+                res_acc[0] =
+                    std::is_same_v<decltype(ref_1), typename AccT::reference>;
+                res_acc[0] &=
+                    std::is_same_v<decltype(ref_2), typename AccT::reference>;
+                if constexpr (!std::is_const_v<T>) {
+                  value_helper::change_val(ref_1, expected_val);
+                  value_helper::change_val(ref_2, changed_val);
+
+                  res_acc[0] &= value_helper::are_equal(ref_1, expected_val);
+                  res_acc[0] &= value_helper::are_equal(ref_2, changed_val);
+
+                  test_accessor_ptr_device(acc, expected_val, res_acc);
+
+                  acc.swap(acc_other);
+                  res_acc[0] &=
+                      value_helper::are_equal(acc_other[0], expected_val);
+                  res_acc[0] &=
+                      value_helper::are_equal(acc_other[1], changed_val);
+                }
+              });
+            })
+            .wait_and_throw();
+      }
+      CHECK(res);
```

Comment:
Minor: it's a pity that we are missing separate log messages for different types of failures. There would be no way to tell which device-side check resulted in failure based on `CHECK(res)` output.